vllm/v1/engine/core_client.py

```diff
@@ -197,28 +201,72 @@ def pin_lora(self, lora_id: int) -> bool:
         return self.engine_core.pin_lora(lora_id)
 
 
+class CoreEngine:
```

Comment:
qq:

I felt like this is a bit confusing given that we have `EngineCore` from the "core" engine, and this `CoreEngine` acts like DPEngineCore iiuc?

Reply:
@aarnphm yes I don't disagree .. I'll think about how to rename/simplify. This class represents one of the engines from the client side so it's not the engine itself. It wraps the process management for that engine for the local engines (which is the case for all of them currently since only single-node is done) but am planning to generalize such that there will be subclasses for local and remote.

Reply:
makes sense. from the looks of it, it acts as a EngineCoreProxy?

Reply:
It's a proxy in some sense but not in the sense that you would think of when you see "proxy" :)